open-ibc/ibc-sdk
# Add Deneb ready images

This PR adds Deneb ready docker images for polymer, prysm and eth-relayer.

The stack is still using Capella blocks though. In order to use Deneb, we need a non-stable `geth`
image which will be added when it becomes stable. Also, the Deneb fork is still being worked on so it
makes sense to wait until it is ready. 

In case you want to test the stack using Deneb blocks, do the following.

1. Pull the latest `geth` image:

```sh
$ docker pull ethereum/client-go:latest
```

2. Tell the `ibc-sdk` to use the image you just pulled

```sh
$ export ETH_DOCKER_IMAGE_TAG=latest
```

3. Adjust the Deneb epoch in the ibc-sdk

```sh
$ sed -ri '/^DENEB_FORK_EPOCH/ s/[0-9]+/2/g' src/lib/chains/eth_helper.ts
```

4. Run the `e2e` test. This should pass by the way

```sh
$ make test-e2e
```

5. Check that the eth-relayer is using Deneb blocks

```sh
$ ibctl -w /tmp/ibctl-tests-cxKlND logs eth-relayer | grep -m2 version
16:23:29.629 INF Got bootstrap version 'deneb'
16:23:34.226 DBG [LQ]: fill hole case, polymer update client,height=24 root=d4971863a83758079574c15772db305d1c57d17060108aed8dce3d6b5f28c651 version=DENEB finalized_height=0 finalized_root= has_next_sync_committee=false
```
